cmd/vulcan-gitleaks/Dockerfile

```diff
@@ -0,0 +1,12 @@
+# Copyright 2019 Adevinta
+
+FROM zricethezav/gitleaks
```

Comment:
I agree with your concern, I think any of your proposed solutions are good!

Reply:
This is a very interesting topic that deserves a discussion. We are using the same approach currently used in this check in some other checks and probably in some other places different from Vulcan checks.

I think the problem in this case is that we are using the default rules they provide, and [that file]((https://github.com/zricethezav/gitleaks/blob/master/config/gitleaks.toml)) is included in the docker image itself. As discussed in another comment below, we might not use their rules, but then we need to be on top of that (operational work) and this is not working very well for us.

So another option would be to pin a version and then get the default ruleset file dynamically or during build time.

Do you think, given that this is not the only place where this is happening, we might merge with this current approach and discuss about this in order to find a solution that works in general?

Reply:
In this case, as a intermediate solution, we could pin the gitleaks Docker image version [configure Dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#docker-registry) to check if there's a new version of the image available.
The problem I see with this solution is that we might be spammed with PRs, since [gitleaks has been creating a new release](https://github.com/zricethezav/gitleaks/releases) almost every week.